modules/jfx.incubator.richtext/src/main/java/jfx/incubator/scene/control/richtext/model/BasicTextModel.java

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright (c) 2023, 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package jfx.incubator.scene.control.richtext.model;
+
+import java.util.ArrayList;
+import java.util.function.Supplier;
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.SimpleBooleanProperty;
+import javafx.scene.layout.Region;
+import jfx.incubator.scene.control.richtext.StyleResolver;
+import jfx.incubator.scene.control.richtext.TextPos;
+
+/**
+ * A StyledTextModel based on plain text paragraphs.
+ * <p>
+ * This class provides no styling.  Subclasses might override {@link #getParagraph(int)} to provide
+ * syntax highlighting based on the model content.
+ * <p>
+ * This model supports custom content storage mechanism via {@link BasicTextModel.Content}.  By default,
+ * the model provides an in-memory storage via its {@link BasicTextModel.InMemoryContent} implementation.
+ */
+public class BasicTextModel extends StyledTextModel {
+    /**
+     * This interface describes the underlying storage mechanism for the BasicTextModel.
+     */
+    public interface Content {
+        /**
+         * Returns the number of paragraphs in this content.
+         * @return number of text lines
```

Comment:
Method description says it returns number of paragraphs but return value says it returns lines. What about document with multi-line paragraphs? What will the return value be?

Reply:
it's the number of paragraphs.